core/src/main/java/jenkins/util/SystemProperties.java

```diff
@@ -0,0 +1,227 @@
+
+    /**
+     * Gets the system property indicated by the specified key.
+     * This behaves just like <code>System.getProperty(String)</code>, except that it
+     * also consults the <code>ServletContext</code>'s "init" parameters.
+     * 
+     * @param      key   the name of the system property.
+     * @return     the string value of the system property,
+     *             or <code>null</code> if there is no property with that key.
+     *
+     * @exception  SecurityException  if a security manager exists and its
+     *             <code>checkPropertyAccess</code> method doesn't allow
+     *              access to the specified system property.
+     * @exception  NullPointerException if <code>key</code> is
+     *             <code>null</code>.
+     * @exception  IllegalArgumentException if <code>key</code> is empty.
+     */
+    public static String getString(String key) {
+        String value = System.getProperty(key); // keep passing on any exceptions
+        if (value != null) {
+            if (LOGGER.isLoggable(Level.INFO)) {
+                LOGGER.log(Level.INFO, "Property (system): {0} => {1}", new Object[] {key, value});
+            }
+        } else if (theContext != null) {
+            value = theContext.getInitParameter(key);
+            if (value != null) {
+                if (LOGGER.isLoggable(Level.INFO)) {
+                    LOGGER.log(Level.INFO, "Property (context): {0} => {1}", new Object[] {key, value});
+                }
+            }
+        } else {
+            if (LOGGER.isLoggable(Level.INFO)) {
+                LOGGER.log(Level.INFO, "Property (not found): {0} => {1}", new Object[] {key, value});
+            }
+        }
+        return value;
+    }
+
+    /**
+     * Gets the system property indicated by the specified key, or a default value.
+     * This behaves just like <code>System.getProperty(String,String)</code>, except
+     * that it also consults the <code>ServletContext</code>'s "init" parameters.
+     * 
+     * @param      key   the name of the system property.
+     * @param      def   a default value.
+     * @return     the string value of the system property,
+     *             or <code>null</code> if there is no property with that key.
+     *
+     * @exception  SecurityException  if a security manager exists and its
+     *             <code>checkPropertyAccess</code> method doesn't allow
+     *              access to the specified system property.
+     * @exception  NullPointerException if <code>key</code> is
+     *             <code>null</code>.
+     * @exception  IllegalArgumentException if <code>key</code> is empty.
+     */
+    public static String getString(String key, String def) {
+        String value = System.getProperty(key); // keep passing on any exceptions
+        if (value != null) {
+            if (LOGGER.isLoggable(Level.INFO)) {
+                LOGGER.log(Level.INFO, "Property (system): {0} => {1}", new Object[] {key, value});
```

Comment:
This may well be true, but I have used this myself and it has produced rather useful output. I'm going to add a comment into the code to that effect so future generations won't bang their head if their property isn't logged.